### ISSUE TITLE
Tool Classes

### ISSUE THREAD

**Issue by ypriverol** [ga4gh/tool-registry-service-schemas#61]
Dear All: 

As you may know, BioContainers is implementing the Ga4Gh API. We have well organized tools and their corresponding containers. We would like to extend some of the definitions of the current API. For example, would be great to have to extend the definition of tool classes. It would be great if we have an example of Tool Classes in the documentation. For example, we see three main classes in BioContainers: 

- Tool: A non annotated tool in biocontainers. I said not annotated because we don't have a proper descriptor file associated with it (CWLTool). We want to use this generic category for all the tools in BioContainers and if the tool gets annotated with the corresponding CWL then it will be updated to CWLTool. 

- Multi-Tool: This category is used for containers "tools" that contains multiple tools inside. We think this class is really useful because our users are interested in using containers with multiple tools. Also we have a mechanism in place to create multi-tools containers automatically. 

- Service: I know most of the things deploy in Ga4GH are workflows but biocontainers is moving forward some services. We will be interested to create a Service class.

In our implementation we will provide the following classes:

```json 
[
  {
    "description": "CommandLineTool",
    "id": "0",
    "name": "CommandLineTool"
  },
  {
    "description": "Workflow",
    "id": "1",
    "name": "Workflow"
  },
{
    "description": "Tool",
    "id": "2",
    "name": "Tool"
  }, 
{
    "description": "Multi-Tool",
    "id": "3",
    "name": "Multi-Tool"
  },
{
    "description": "Service",
    "id": "4",
    "name": "Service"
  }
]
```
Is this possible @denis-yuen @bwalsh

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TRS-2)
┆Project Name: GA4GH tool-registry-service
┆Issue Number: TRS-2

**Comment by denis-yuen** [comment]
Yes,
So the Dockstore implementation of TRS splits things up similarly and we may go down the services route too (let's compare notes!). 

But for now, we have 
```
[
  {
    "description": "CommandLineTool",
    "id": "0",
    "name": "CommandLineTool"
  },
  {
    "description": "Workflow",
    "id": "1",
    "name": "Workflow"
  }
]
```

I think putting our examples in the documentation would be a fine idea and perhaps lay the foundation for future standardization.

**Comment by ypriverol** [comment]
Well by definition we can use other classifications?

**Comment by denis-yuen** [comment]
I'm not sure I understand the question, could you rephrase?

**Comment by ypriverol** [comment]
What I mean is , Can we extend the current classifications?

**Comment by denis-yuen** [comment]
Oh yes, there are no current restrictions on tool classes, new ones can be added. 
I just meant it would be nice to document them here (in this repo) so different teams have the option of  using the same classes to mean the same thing.

**Comment by denis-yuen** [comment]
(question seems answered)
